js/views/userPageVw.js

```diff
@@ -595,6 +607,11 @@ module.exports = Backbone.View.extend({
 
   storeClick: function(e){
     "use strict";
+    var select = document.getElementById('categories');
+    if (select.value != "all"){
+        select.value = "all"
+        this.categoryChanged();
```

Comment:
You have a change event on the select that fires when its value is changed, and triggers this.categoryChanged();

This code also changes the value of the select, I think it might result in triggering this.categoryChanged(); twice.